feat: Create PayButton API Route integrated with MySQL database

alcipir opened the pull request:
Description: 
- Adds a makefile, update REAMDE, fix docker-compose, add sequelize (to connect to mysql through NextJS) and required config, including initial models and migrations. 
- Adds integration with MySQL database for creating paybuttons through the API Route /api/paybutton

Test Plan: run `yarn` to update dependencies,  `make dev` to run the application and mysql, `make init-db` to run migrations and seeds. Every step should be successful.
Open up localhost:3000, then login and click on `Add PayButton`. On the browser console, should show a 201 response (successfully created)

Gattaca1 commented:
As David suggested, moving this comment from Telegram to here
![Screenshot from 2022-04-09 21-59-43](https://user-images.githubusercontent.com/7704588/163459835-c9582288-7b4b-450e-bce3-e6ae5d6fc9f0.png)

are you using `sudo` to run make?
i have a permission error i haven't bothered to track down yet when running `make dev`, but `sudo make dev` fixes it
theres a lot of different things happening with this branch, i'm not sure which change is the issue

alcipir commented:
Updates: 

- Removed init-db step, now `make dev` should build and run the scripts necessary to have the paybutton server working
- Added instructions on how to get testing credentials to add to .env.local

Gattaca1 commented:
Could you add testing credentials from supertoken to .env?

Also I'm still getting an error here 

![Screenshot from 2022-04-20 22-32-12](https://user-images.githubusercontent.com/7704588/164383424-3f38d340-3921-4872-9dc7-e333c5452d56.png)

alcipir commented:
Just pushed a fix that only show social logins as long as you provide some credentials on .env file. (So it shouldn't break your app if you don't have it) 

Let me know of any more issues, and if needed testing credentials, hit me up on telegram @Gattaca1

Klakurka commented on .env:
```diff
@@ -1 +1,2 @@
-GRPC_NODE_URL=bchd.greyh.at:8335
\ No newline at end of file
+GRPC_NODE_URL=bchd.greyh.at:8335
```
Where did this URL come from? We have our own BCHD node.

Klakurka replied:
Where did this URL come from? We have our own BCHD node.

Klakurka commented on README.md:
`````diff
@@ -4,7 +4,24 @@ https://paybutton.org
 ## Developing
 - [Install and configure Docker](https://docs.docker.com/get-docker/)
-- Run the following make command to build the docker image and run the server locally:
+- [Install docker-compose](https://docs.docker.com/compose/install/)
+- Fill up .env or .env.local file with all social provider credentials. You can get testing credentials from [here](https://supertokens.com/docs/thirdpartyemailpassword/quick-setup/backend#2-initialise-supertokens).
+Your .env or .env.local file should look like this:
+````
+APPLE_CLIENT_ID=<insert credentials here>
+APPLE_KEY_ID=<insert credentials here>
+APPLE_PRIVATE_KEY=<insert credentials here>
+APPLE_TEAM_ID=<insert credentials here>
+FACEBOOK_CLIENT_ID=<insert credentials here>
+FACEBOOK_CLIENT_SECRET=<insert credentials here>
+GITHUB_CLIENT_ID=<insert credentials here>
+GITHUB_CLIENT_SECRET=<insert credentials here>
+GOOGLE_CLIENT_SECRET=<insert credentials here>
+SUPERTOKENS_API_KEY=<insert credentials here>
+SUPERTOKENS_CONNECTION_URI=<insert credentials here>
+```
+- **If you run docker as root user, run the following commands with `sudo`** 
+- Run the following make command to build/pull the relevant docker images and run the server locally:
`````
Doesn't look related to what the branch is supposed to be so should be committed separately.

Klakurka replied:
Doesn't look related to what the branch is supposed to be so should be committed separately.

Klakurka commented on config/frontendConfig.tsx:
```diff
@@ -3,6 +3,20 @@ import SessionReact from 'supertokens-auth-react/recipe/session'
 import { appInfo } from './appInfo'
 
 export let frontendConfig = () => {
+  const getAvailableSocialProviders = () => {
+    const socialProviderNodes = {
+      'github': process.env.GITHUB_CLIENT_ID && ThirdPartyEmailPasswordReact.Github.init(),
+      'google': process.env.GOOGLE_CLIENT_ID && ThirdPartyEmailPasswordReact.Google.init(),
+      'facebook': process.env.FACEBOOK_CLIENT_ID && ThirdPartyEmailPasswordReact.Facebook.init(),
+      'apple': process.env.APPLE_CLIENT_ID && ThirdPartyEmailPasswordReact.Apple.init(),
+    }
+    const availableSocialProvidersNodes = Object.keys(socialProviderNodes).map(providerKey => {
+      if (socialProviderNodes[providerKey] !== undefined)
+        return socialProviderNodes[providerKey]
+    })
+    return availableSocialProvidersNodes.filter(provider => provider !== undefined)
+  }
+   
```
Goes for all the social login stuff (should be in a different branch).

Klakurka replied:
Goes for all the social login stuff (should be in a different branch).

Klakurka commented on docker-compose.yml:
```diff
@@ -1,21 +1,24 @@
 services:
   paybutton:
+    depends_on: 
+      - mysql
```
Adding in MySQL should have been its own branch, separated from the "add a paybutton" stuff.

https://ourmachinery.com/post/step-by-step-programming-incrementally/

Klakurka replied:
Adding in MySQL should have been its own branch, separated from the "add a paybutton" stuff.

https://ourmachinery.com/post/step-by-step-programming-incrementally/

Gattaca1 commented:
There are a ton of features in this branch, lets break it down some so its easier to deconstruct

Updating the README can happen in any branch to reflect changes, but just about everything else should have its own branch

- Adds a makefile & update REAMDE, 

- fix docker-compose, 
- Dependency changes,
- Add MySQL,

- add sequelize (to connect to mysql through NextJS) and required config, including initial models and migrations.

- Adds integration with MySQL database for creating paybuttons through the API Route /api/paybutton,
- social login changes,
